File: src/main/java/com/hy/http/file/FileBinary.java
package com.hy.http.file;

import android.text.TextUtils;

import com.hy.frame.util.FileUtil;
import com.hy.frame.util.MyLog;

import java.io.File;

public class FileBinary implements Binary {

    private File file;
    private String fileName;
    private String mimeType;
    public static final String MIME_TYPE_FILE = "application/octet-stream";

    public FileBinary(File file) {
        this(file, file.getName());
    }

    public FileBinary(File file, String fileName) {
        this(file, fileName, null);
    }

    public FileBinary(File file, String fileName, String mimeType) {
        if (file == null) {
            MyLog.w("File == null");
        } else if (!file.exists()) {
            MyLog.w("File isn't exists");
        }
        this.file = file;
        this.fileName = fileName;
        this.mimeType = mimeType;
    }


    @Override
    public File getFile() {
        return file;
    }

    @Override
    public long getLength() {
        if (file == null || !file.exists())
            return 0;
        return this.file.length();
    }

    @Override
    public String getFileName() {
        return fileName;
    }

    @Override
    public String getMimeType() {
        if (TextUtils.isEmpty(mimeType)) {
            mimeType = FileUtil.getMimeTypeByUrl(file.getAbsolutePath());
            if (TextUtils.isEmpty(mimeType))
                mimeType = MIME_TYPE_FILE;
        }
        return mimeType;
    }
}

File: src/main/java/com/hy/http/file/FileBinary.kt
package com.hy.http.file

import com.hy.frame.util.FileUtil
import java.io.File
import java.io.IOException

class FileBinary(override val file: File, fileName: String? = file.name, mimeType: String? = null) : Binary {

    override val fileName: String
    override val mimeType: String
    override val length: Long

    companion object {
        val MIME_TYPE_FILE = "application/octet-stream"
    }

    init {
        if (!file.exists()) {
            throw IOException("File isn't exists")
        }
        this.length = file.length()
        this.fileName = fileName!!
        this.mimeType = if (mimeType.isNullOrEmpty()) FileUtil.getMimeTypeByUrl(file.absolutePath) else MIME_TYPE_FILE
    }
}
